app/rest/rest/src/main/java/io/syndesis/rest/util/ReflectiveFilterer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.rest.util;
+
+import io.syndesis.model.ListResult;
+import io.syndesis.rest.v1.util.PredicateFilter;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class ReflectiveFilterer<T> implements Function<ListResult<T>, ListResult<T>> {
+
+    private final List<PredicateFilter<T>> predicateFilters;
+
+    public ReflectiveFilterer(Class<T> modelClass, List<FilterOptionsParser.Filter> filters) {
+        predicateFilters = new ArrayList<>(filters.size());
+        for (FilterOptionsParser.Filter f : filters) {
+            String op = f.getOperation().filter("="::equals).orElseThrow(
```

Comment:
❤️ that `"="::equals` method statement.  Slowly we are getting into Perl and Scala obfuscation moody waters. I knew, Java would end up there.

btw, what's so bad about 

```java
String op = f.getOperation();
if (! "=".equals(op)) {
    throw new IllegalArgumentException("Unknown filter operation " + op);
}
```

Reply:
For more ops to come, 

```java
Set<String> opsAllowed = new HashSet<>(Arrays.asList("=", "!="));
String op = f.getOperation();
if (!opsAllowed.contains("op")) {
    throw new IllegalArgumentException("Unknown filter operation " + op);
}
```

Reply:
See also https://github.com/syndesisio/syndesis/pull/495#discussion_r154291899

It's maybe because I've written tons of 'obfuscated' code myself and have seen my favourite programming language declined because people couldn't deal with its power, that I'm somewhat sensitive here. But at the end, it's probably only my opinion, that I don't want to stop while reviewing and thinking about a line for longer than a minute just to honour its intellectual beauty (it **is** beautiful btw ;-)

Sorry, became a pragmatist over the years ....

Reply:
I like the functional approach myself but OK...